src/ray/protobuf/autoscaler.proto

```diff
@@ -0,0 +1,172 @@
+autoscalers.
+
+It is currently unstable and under active development, but it is
+intended to become stable and forwards-compatible in the future.
+*/
+
+
+syntax = "proto3";
+option cc_enable_arenas = true;
+package ray.rpc;
+
+message ResourceBundle {
+  // The fixed point resource size of the resource (granularity: 1/10000).
+  map<string, int64> resources = 1;
+}
+
+// Type of placement group strategy.
+enum PlacementStrategy {
+  // Packs Bundles into as few nodes as possible.
+  PACK = 0;
+
+  // Places Bundles across distinct nodes or processes as even as possible.
+  SPREAD = 1;
+
+  // Packs Bundles within one node. The group is not allowed to span multiple nodes.
+  STRICT_PACK = 2;
+
+  // Places Bundles across distinct nodes.
+  // The group is not allowed to deploy more than one bundle on a node.
+  STRICT_SPREAD = 3;
+}
+
+enum SourceType {
+  // A task that (assuming no deadlock can eventually be scheduled on the cluster.
+  TASK = 0;
+
+  // A actor that (assuming no deadlock can eventually be scheduled on the cluster.
+  ACTOR = 1;
+
+  // A placement group that (assuming no deadlock can eventually be scheduled on the cluster.
+  PLACEMENT_GROUP = 2;
+
+  // Represents a job's call to
+  // `ray.autoscaler.sdk.request_resources()`. This type requests
+  // resources to be available on the cluster but won't actually
+  // utilize them.
+  SDK_RESOURCE_REQUEST = 3;
+}
+
+// A workload (task/actor/placement group) unfulfilled by the the cluster's internal scheduler(s).
+message QueuedResources {
+  // The number of resources of this quantity. This should be
+  // functionally equivalent to setting quantity=1 and repeating the
+  // field many times.
+  uint32 quantity = 1;
+
+  // Whether the queued resources can eventually be scheduled on the cluster
+  // (neglecting deadlock).
+  bool is_feasible = 2;
+
+  // The source of this queued resource.
+  SourceType type = 3;
+
+  // If the queued resource represents an actor or task, there is
+  // 1 entry in `bundles`. In a placement group, each bundle
+  // receives its own entry.
+  repeated ResourceBundle bundles = 4;
+
+  // The placement strategy of bundles within this queued resources.
+  PlacementStrategy placement_strategy = 5;
+}
+
+message GetAllQueuedResourcesRequest {
+}
+
+message GetAllQueuedResourcesResponse {
+  // The set of all queued resources.
+  repeated QueuedResources queued_resources = 1;
+}
+
+message NodeInfo {
+  // State of a node.
+  enum NodeState {
+    // The node has not joined the cluster yet.
+    PENDING = 0;
+
+    // The node has joined the cluser and is healthy.
+    ALIVE = 1;
+
+    // The node is marked for removal, but it may not be safe to remove it yet.
+    DRAINING = 2;
+
+    // The node is no longer active in the cluster it is safe to remove it.
+    DEAD = 3;
+  }
+
+  // The state of the node.
+  NodeState node_state = 1;
+
+  // TODO: Ideally we unify GCS's node id with the autoscaler's here.
```

Comment:
Cloud provider identity of a node is always going to be a different thing.
But we could probably improve how the two identities are managed.

Reply:
> Cloud provider identity of a node is always going to be a different thing.

Or maybe not -- SetNodeInfo seems a fundamental change to the interface.
Looks like the autoscaler is now telling the GCS what the name of the node is, so we can pass the cloud provider name of the node straight to the the GCS?

Reply:
err...or maybe there's a typo below

Reply:
Disregard above and see question about NodeState and NodeInfo below.

Reply:
Eh, or don't disregard.
It looks like we have a bit of a hanging question from the original about how exactly `SetNodeState` is supposed to work.
https://docs.google.com/document/d/197P0Ia_dPbHgfvJbo0gauP1BhPrlEheMibrnZjLhOUU/edit?disco=AAAAVYH-xe0